### ISSUE TITLE
wideDeep模型进行CTR预测时，train_loss和valid_loss结果异常

### ISSUE THREAD

**Issue by nielei0128** [RUCAIBox/RecBole#1197]
您好，我在使用WideDeep模型对criteo数据集进行训练时，出现了两个问题:
(1)train_loss远大于valid_loss;
(2)valid_loss在第二个epoch之后开始变大。
请问有什么解决的建议或方法吗？
十分感谢！！！
这是我的配置文件内容：
```
data_path: .\dataset

# dataset config
dataset: 'criteo_sample'
field_separator: "\t"
seq_separator: " "
USER_ID_FIELD: ~
ITEM_ID_FIELD: ~
LABEL_FIELD: label
TIME_FIELD: index

load_col: 
    inter: '*'

fill_nan: True
normalize_all: True


# training and evaluation
show_progress: False
dropout_prob: 0.5
epochs: 500
train_batch_size: 4096
eval_args:
  mode: labeled
  group_by: ~
metrics: ['AUC', 'LogLoss']
valid_metric: LogLoss
stopping_step: 10

# model
model: 'WideDeep'
embedding_size: 100
```
这是模型运行后输出的结果：
```
18 Mar 22:44    INFO  epoch 0 training [time: 18.67s, train loss: 5.8079]
18 Mar 22:44    INFO  epoch 0 evaluating [time: 0.65s, valid_score: 0.593100]
18 Mar 22:44    INFO  valid result: 
auc : 0.5658    logloss : 0.5931
18 Mar 22:44    INFO  Saving current: saved\WideDeep-Mar-18-2022_22-43-58.pth
18 Mar 22:46    INFO  epoch 1 training [time: 106.19s, train loss: 22.2083]
18 Mar 22:46    INFO  epoch 1 evaluating [time: 0.73s, valid_score: 0.521800]
18 Mar 22:46    INFO  valid result: 
auc : 0.716    logloss : 0.5218
18 Mar 22:46    INFO  Saving current: saved\WideDeep-Mar-18-2022_22-43-58.pth
18 Mar 22:48    INFO  epoch 2 training [time: 119.46s, train loss: 19.6216]
18 Mar 22:48    INFO  epoch 2 evaluating [time: 0.75s, valid_score: 0.506700]
18 Mar 22:48    INFO  valid result: 
auc : 0.729    logloss : 0.5067
18 Mar 22:48    INFO  Saving current: saved\WideDeep-Mar-18-2022_22-43-58.pth
18 Mar 22:50    INFO  epoch 3 training [time: 134.15s, train loss: 16.2056]
18 Mar 22:50    INFO  epoch 3 evaluating [time: 1.02s, valid_score: 0.543600]
18 Mar 22:50    INFO  valid result: 
auc : 0.7078    logloss : 0.5436
18 Mar 22:52    INFO  epoch 4 training [time: 129.08s, train loss: 13.0275]
18 Mar 22:52    INFO  epoch 4 evaluating [time: 0.88s, valid_score: 0.596600]
18 Mar 22:52    INFO  valid result: 
auc : 0.6942    logloss : 0.5966
18 Mar 22:54    INFO  epoch 5 training [time: 126.00s, train loss: 11.2008]
18 Mar 22:54    INFO  epoch 5 evaluating [time: 0.86s, valid_score: 0.653400]
18 Mar 22:54    INFO  valid result: 
auc : 0.6888    logloss : 0.6534
18 Mar 22:56    INFO  epoch 6 training [time: 112.20s, train loss: 10.0967]
18 Mar 22:56    INFO  epoch 6 evaluating [time: 0.65s, valid_score: 0.723500]
18 Mar 22:56    INFO  valid result: 
auc : 0.6794    logloss : 0.7235
18 Mar 22:58    INFO  epoch 7 training [time: 94.74s, train loss: 9.4679]
18 Mar 22:58    INFO  epoch 7 evaluating [time: 0.61s, valid_score: 0.791700]
18 Mar 22:58    INFO  valid result: 
auc : 0.6755    logloss : 0.7917
18 Mar 22:59    INFO  epoch 8 training [time: 89.75s, train loss: 9.0178]
18 Mar 22:59    INFO  epoch 8 evaluating [time: 0.59s, valid_score: 0.865600]
18 Mar 22:59    INFO  valid result: 
auc : 0.6701    logloss : 0.8656
18 Mar 23:01    INFO  epoch 9 training [time: 90.20s, train loss: 8.6774]
18 Mar 23:01    INFO  epoch 9 evaluating [time: 0.61s, valid_score: 0.900900]
18 Mar 23:01    INFO  valid result: 
auc : 0.6709    logloss : 0.9009
18 Mar 23:02    INFO  epoch 10 training [time: 91.69s, train loss: 8.4031]
18 Mar 23:02    INFO  epoch 10 evaluating [time: 0.55s, valid_score: 0.990200]
18 Mar 23:02    INFO  valid result: 
auc : 0.6667    logloss : 0.9902
18 Mar 23:04    INFO  epoch 11 training [time: 95.06s, train loss: 8.1371]
18 Mar 23:04    INFO  epoch 11 evaluating [time: 0.59s, valid_score: 1.052300]
18 Mar 23:04    INFO  valid result: 
auc : 0.6651    logloss : 1.0523
18 Mar 23:05    INFO  epoch 12 training [time: 100.63s, train loss: 7.9100]
18 Mar 23:06    INFO  epoch 12 evaluating [time: 0.63s, valid_score: 1.126700]
18 Mar 23:06    INFO  valid result: 
auc : 0.6628    logloss : 1.1267
18 Mar 23:07    INFO  epoch 13 training [time: 101.07s, train loss: 7.7787]
18 Mar 23:07    INFO  epoch 13 evaluating [time: 0.66s, valid_score: 1.159900]
18 Mar 23:07    INFO  valid result: 
auc : 0.6639    logloss : 1.1599
18 Mar 23:07    INFO  Finished training, best eval result in epoch 2
```

**Comment by Sherry-XLL** [comment]
@nielei0128 你好，模型的输出中没有 `valid_loss`，`valid_score` 表示的是当前 epoch 的 logloss 的值，和 train loss 没有可比性。至于 train loss 的第二轮相比第一轮突然变大，可能是模型训练初期不稳定的原因，当前选择的数据集也许较为简单，导致模型在 epoch 2 就到达了最佳状态。

**Comment by nielei0128** [comment]
> @nielei0128 你好，模型的输出中没有 `valid_loss`，`valid_score` 表示的是当前 epoch 的 logloss 的值，和 train loss 没有可比性。至于 train loss 的第二轮相比第一轮突然变大，可能是模型训练初期不稳定的原因，当前选择的数据集也许较为简单，导致模型在 epoch 2 就到达了最佳状态。
那么应当如何输出每一轮的valid loss？wideDeep的loss_func是BCEloss，为什么不能用logloss表示valid loss呢？对于在epoch2就达到最佳状态这个问题，请问在参数设定上有什么建议吗

**Comment by Sherry-XLL** [comment]
@nielei0128 你好，我之前的表述可能有点问题，log loss 确实可以作为模型在验证集上的一种损失衡量，由于 trainer 在计算模型的损失时计算的是一个 epoch 内所有 batch 的 total_loss，训练集和验证集数据量的差异导致了损失在数值上的差异。对于 epoch2 上达到最优结果，直接原因还是数据量太少引起模型过早收敛，可以选择适当改变 batch_size 和 learning_rate 等超参数来调整模型的训练。同时，较为简单的场景不宜设置过大的 embedding_size，适当减小 embedding_size 也会更适合当前的应用。

**Comment by nielei0128** [comment]
> @nielei0128 你好，我之前的表述可能有点问题，log loss 确实可以作为模型在验证集上的一种损失衡量，由于 trainer 在计算模型的损失时计算的是一个 epoch 内所有 batch 的 total_loss，训练集和验证集数据量的差异导致了损失在数值上的差异。对于 epoch2 上达到最优结果，直接原因还是数据量太少引起模型过早收敛，可以选择适当改变 batch_size 和 learning_rate 等超参数来调整模型的训练。同时，较为简单的场景不宜设置过大的 embedding_size，适当减小 embedding_size 也会更适合当前的应用。

明白啦！谢谢！
